package.json

```diff
@@ -30,15 +30,16 @@
     "dev": "node esbuild.mjs --watch",
     "test": "jest",
     "test:watch": "jest --watch",
-    "lint": "tsc --noEmit && eslint \"{src,test,scripts}/**/*.{js,jsx,mjs,ts,tsx}\"",
+    "typecheck": "tsc --noEmit",
+    "lint": "eslint \"{src,test,scripts}/**/*.{js,jsx,mjs,ts,tsx}\"",
```

Comment:
These two should be separate, since a project can be conformed to TS but not to lint rules. Always running type checks when linting is extra effort and prolongs the command execution too.